src/app/dashboard-widgets/environment-widget/environment-widget.component.ts

```diff
@@ -1,232 +1,40 @@
-import { Component, OnDestroy, OnInit, ViewEncapsulation } from '@angular/core';
+import { Component, OnInit, ViewEncapsulation } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 
 import { BehaviorSubject, ConnectableObservable, Observable, Subject, Subscription } from 'rxjs/Rx';
 
-import { Context, Contexts } from 'ngx-fabric8-wit';
+import { Context, Contexts, Spaces } from 'ngx-fabric8-wit';
 
-import {
-  Environment, Space, SpaceStore
-} from '../../../a-runtime-console/index';
+import { Environment as ModelEnvironment } from '../../../app/space/create/deployments/models/environment';
 
-// non-exported classes from runtime that need to be refactored
-import { DeploymentView, DeploymentViews } from '../../../a-runtime-console/index';
-import { AppDeployments, AppEnvironmentDetails, EnvironmentDeployments } from '../../../a-runtime-console/index';
-import { AbstractWatchComponent } from '../../../a-runtime-console/index';
-import { DeploymentService } from '../../../a-runtime-console/index';
-import { ServiceService } from '../../../a-runtime-console/index';
-import { DeploymentConfigService } from '../../../a-runtime-console/index';
-import { RouteService } from '../../../a-runtime-console/index';
+import {
+  ApplicationAttributesOverview,
+  DeploymentsService,
+  Space
+} from '../../../app/space/create/deployments/services/deployments.service';
 
 @Component({
   encapsulation: ViewEncapsulation.None,
   selector: 'fabric8-environment-widget',
   templateUrl: './environment-widget.component.html',
-  styleUrls: ['./environment-widget.component.less']
+  styleUrls: ['./environment-widget.component.less'],
+  providers: [DeploymentsService]
 })
-export class EnvironmentWidgetComponent extends AbstractWatchComponent  implements OnInit, OnDestroy {
+export class EnvironmentWidgetComponent implements OnInit {
 
-  loading: Subject<boolean> = new BehaviorSubject(true);
+  spaceId: Observable<string>;
+  appInfos: Observable<ApplicationAttributesOverview[]>;
```

Comment:
appInfos still needs to be assigned as it was in the last version of this commit, so the spaceId still needs to be subscribed on - the returned subscription just doesn't need to be tracked.